sqlx-sqlite/src/query_result.rs

```diff
@@ -24,3 +24,13 @@ impl Extend<SqliteQueryResult> for SqliteQueryResult {
         }
     }
 }
+
+#[cfg(feature = "any")]
+impl From<SqliteQueryResult> for crate::any::AnyQueryResult {
+    fn from(done: SqliteQueryResult) -> Self {
+        crate::any::AnyQueryResult {
+            rows_affected: done.rows_affected(),
+            last_insert_id: Some(done.last_insert_rowid()),
```

Comment:
`last_insert_rowid()` is 0 if no qualifying `INSERT`s have occurred on the database connection: https://www.sqlite.org/c3ref/last_insert_rowid.html

I'd map 0 to `None` here.

We could have surfaced this better as something like `Option<NonZeroI64>` but hindsight is 20/20.

Reply:
[rewrite use and address implementation concerns](https://github.com/launchbadge/sqlx/pull/3608/commits/b5c279abedf0a82741b7f1802e17479ca2a64f3b)